src/lib/adapters/pubsub.ts

```diff
@@ -33,26 +33,29 @@ export const fuQuPubSub: FuQuCreator<FuQuPubSubOptions, Message> = (pubSub: PubS
             }
             return x[0];
         });
-    const subscription = topic.then(topic =>
-        topic
-            .subscription(topicName, options?.subscriptionOptions)
-            .get({ autoCreate: true })
-            .then(x => {
-                if (options?.subscriptionOptions) {
-                    x[0].setOptions(options?.subscriptionOptions);
-                }
-                return x[0];
-            })
-    );
+    let subscription: Promise<Subscription> | undefined;
+
     return createFuQu(
         {
             name: 'pubsub',
-            isAlive: () => subscription.then(s => s.isOpen),
-            close: () => subscription.then(s => s.close()),
+            isAlive: () =>
+                subscription ? subscription.then((s) => s.isOpen) : topic.then((t) => t.get()).then((t) => !!t[0]),
+            close: () => (subscription ? subscription.then((s) => s.close()) : Promise.resolve()),
             publishJson: async (payload, attributes) => {
                 await (await topic).publishJSON(payload, attributes);
             },
-            registerHandler: async handler => {
+            registerHandler: async (handler) => {
+                subscription = topic.then((t) =>
```

Comment:
This way we are creating a new subscription for each `handler` registration. I know it will probably get the instance from the remote, but why bother when we already have the subscription instance from the first call. What do you think?

Reply:
Fixed in eaa72e2

Reply:
Looks good